openfang_config/upstream-overrides/crates/openfang-api/static/js/pages/chat.js

```
@@ -0,0 +1,1274 @@
+              typingMsg.text = 'Using ' + (data.tool || 'tool') + '...';
+              this.messages.splice(toolTypIdx, 1, typingMsg);
+            }
+            this._resetTypingTimeout();
+          } else if (data.state === 'stop') {
+            this._clearTypingTimeout();
+          }
+          break;
+
+        case 'phase':
+          // Show tool/phase progress so the user sees the agent is working
+          let phaseIdx = this.messages.length - 1;
```

Biome (2.4.10)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)